serverLib.sml

```diff
@@ -667,17 +667,34 @@ local
   open HTML
 in
 
+  fun cakeml_commit_link s =
+    a (String.concat[cakeml_github,"/commit/",s]) s
+  fun hol_commit_link s =
+    a (String.concat[hol_github,"/commit/",s]) s
+  fun cakeml_pr_link ss =
+    a (String.concat[cakeml_github,"/pull/",Substring.string(Substring.triml 1 ss)]) (Substring.string ss)
+
+  fun escape_char #"<" = "&lt;"
+    | escape_char #">" = "&gt;"
+    | escape_char c = if Char.isPrint c orelse Char.isSpace c then String.str c else ""
+  val escape = String.translate escape_char
+
   fun job_link q id =
     let
       val jid = Int.toString id
       val f = OS.Path.concat(q,jid)
       val inp = TextIO.openIn f
-      val typ = read_job_type inp
+      val pr_info = read_job_pr inp
                 handle IO.Io _ => cgi_die 500 ["cannot open ",f]
                      | Option => cgi_die 500 [f," has invalid file format"]
       val worker = read_job_worker inp
-      val format_type = if q = "finished" then span (status_attrs (read_status inp)) else String.concat
+      val status = read_status inp
       val () = TextIO.closeIn inp
+      val format_status = if q = "finished" then span (status_attrs status) else String.concat
+      val typ_string =
+        case pr_info of NONE => format_status ["master"]
+                      | SOME (pr, branch) => String.concat[format_status [cakeml_pr_link pr],
+                                                           escape (Substring.string branch)]
```

Comment:
I think it looks a bit too long having both PR number and branch in the same column, specially since some branch names are horrendously long. I suggest we only display the branch name with a link to the relevant PR (like this: [pr-branch-names](https://github.com/CakeML/regression/pull/20)) I guess the main purpose of the "type" column/field  is the inform at a glance what  is the change being tested, and I think the branch names is more than enough to accomplish this = )

Reply:
Sounds good. And we can change the column title to 'branch'.